src/main/resources/dotty_res/scripts/hljs-scala3.js

```diff
@@ -0,0 +1,415 @@
+      hljs.QUOTE_STRING_MODE,
+      NUMBER,
+      PROBABLY_TYPE
+    ]
+  }
+
+  // (using T1, T2, T3)
+  const CTX_PARAMS = {
+    className: 'cparams',
+    begin: /\(using (?!\w+:)/, end: /\)/,
+    excludeBegin: false,
+    excludeEnd: true,
+    relevance: 5,
+    keywords: {
+      $pattern: alwaysKeywords.$pattern,
+      keyword: 'using ?=> =>> => _ ? <: >:',
+      literal: alwaysKeywords.literal,
+      built_in: alwaysKeywords.built_in
+    },
+    contains: [
+      PROBABLY_TYPE
+    ]
+  }
+
+  // String interpolation
+  const SUBST = {
+    className: 'subst',
+    variants: [
+      {begin: /\$[a-zA-Z_]\w*/},
+      {
+        begin: /\${/, end: /}/,
+        contains: [
+          NUMBER,
+          hljs.QUOTE_STRING_MODE
+        ]
+      }
+    ]
+  }
+
+  const STRING = {
+    className: 'string',
+    variants: [
+      hljs.QUOTE_STRING_MODE,
+      {
+        begin: '"""', end: '"""',
+        contains: [hljs.BACKSLASH_ESCAPE],
+        relevance: 10
+      },
+      {
+        begin: alphaId.source + '"', end: '"',
+        contains: [hljs.BACKSLASH_ESCAPE, SUBST],
+        illegal: /\n/,
+        relevance: 5
+      },
+      {
+        begin: alphaId.source + '"""', end: '"""',
+        contains: [hljs.BACKSLASH_ESCAPE, SUBST],
+        relevance: 10
+      }
+    ]
+  }
+
+  // Class or method apply
+  const APPLY = {
+    begin: /\(/, end: /\)/,
+    excludeBegin: true, excludeEnd: true,
+    keywords: {
+      $pattern: alwaysKeywords.$pattern,
+      keyword: 'using ' + alwaysKeywords.keyword,
+      literal: alwaysKeywords.literal,
+      built_in: alwaysKeywords.built_in
+    },
+    contains: [
+      STRING,
+      NUMBER,
+      hljs.C_BLOCK_COMMENT_MODE,
+      PROBABLY_TYPE,
+    ]
+  }
+
+  // @annot(...) or @my.package.annot(...)
+  const ANNOTATION = {
+    className: 'meta',
+    begin: `@${id.source}(\\.${id.source})*`,
+    contains: [
+      APPLY,
+      hljs.C_BLOCK_COMMENT_MODE
+    ]
+  }
+
+  // Documentation
+  const SCALADOC = hljs.COMMENT('/\\*\\*', '\\*/', {
+    contains: [
+      {
+        className: 'doctag',
+        begin: /@[a-zA-Z]+/
+      },
+      // markdown syntax elements:
+      {
+        className: 'code',
+        variants: [
+          {begin: /```.*\n/, end: /```/},
+          {begin: /`/, end: /`/}
+        ],
+      },
+      {
+        className: 'bold',
+        variants: [
+          {begin: /\*\*/, end: /\*\*/},
+          {begin: /__/, end: /__/}
+        ],
+      },
+      {
+        className: 'emphasis',
+        variants: [
+          {begin: /\*(?![\*\s/])/, end: /\*/},
+          {begin: /_/, end: /_/}
+        ],
+      },
+      {
+        className: 'bullet', // list item
+        begin: /- (?=\S)/, end: /\s/,
+      },
+      {
+        className: 'link',
+        begin: /(?<=\[.*?\])\(/, end: /\)/,
+      }
+    ]
+  })
+
+  // Methods
+  const METHOD = {
+    className: 'function',
+    begin: /(transparent +)?(inline +)?def/, end: / =|\n/,
```

Comment:
If I guess correctly how this grammar works, this should be `((transparent|inline) +)*`, so that the order of modifiers doesn't matter.

I'm also not sure if definitions like `inline protected def foo` would be highlighted properly, we may to instead have a regex like `(protected|private|final|override|transparent|inline)` here.

Note that if those definitions won't work, we also need to modify the `TYPEDEF` definition similarly. A good way to do that would probably be by storing and reusing the common sub-regex.

Reply:
You’re right! I’ll fix that.